core/src/main/java/hudson/slaves/NodeProvisioner.java

```diff
@@ -817,4 +828,81 @@ private static float getFloatSystemProperty(String propName, float defaultValue)
             }
         return defaultValue;
     }
+
+    private static void fireOnFailure(final NodeProvisioner.PlannedNode plannedNode, final Throwable cause) {
+        for (CloudProvisioningListener cl : CloudProvisioningListener.all()) {
+            try {
+                cl.onFailure(plannedNode, cause);
+            } catch (Throwable e) {
+                LOGGER.log(Level.SEVERE, "Unexpected uncaught exception encountered while "
+                        + "processing onFailure() listener call in " + cl + " for agent "
+                        + plannedNode.displayName, e);
+                if (e instanceof Error) {
```

Comment:
🐜 De-facto we process errors in the logging handler. E.g. logging constructs the string, and it may cause OutOfMemoryError, which suppresses the original one. I'd rather vote that Errors are being rethrown without any processing